### ISSUE TITLE
Rotarycraft Cold Reservoirs freeze Streams mod water into vanilla ice instead of Streams ice

### ISSUE THREAD

**Issue by Barhandar** [ReikaKalseki/Reika_Mods_Issues#787]
This causes the streams, which normally have a current, to lose it once ice thaws. Specifically Gelid Cryotheum was used to make reservoir cold. 

Also, water and fluids of its temperature (300K degrees) insta-melt ice around the Reservoir.

**Comment by TomeWyrm** [comment]
Those behaviors are intentional. Though the incorrect ice is likely a mod interaction bug or oversight.

**Comment by ReikaKalseki** [comment]
Streams has ice?

**Comment by Builderb0y** [comment]
Yes, 17 different block IDs for them, all with 16 metadatas; just like its custom water blocks. ID stores the direction the water is flowing, and metadata stores how full the block is. Shouldn't be hard to map each custom water block to a custom ice block, preserving the metadata.

**Comment by ReikaKalseki** [comment]
It is going to be an enormous pain, actually...

**Comment by Barhandar** [comment]
Given Streams is written in Scala and not Java, yeah, sounds like a pain. If anything, [this seems to be the code in it](https://github.com/delvr/Streams/blob/master/src/main/scala/streams/block/BlockRiver.scala#L124) that handles "natural" freezing.

**Comment by ReikaKalseki** [comment]
...I cannot even invoke that, as it would fail in non-cold or sealed areas...

**Comment by Barhandar** [comment]
But you can poke at `setBlockAt(xyz, FixedFlowBlock(Material.ice, dx, dz))` since it's the function that places "proper" ice.

**Comment by ReikaKalseki** [comment]
I suppose, yes. Still, adding reflective invocation - which even the block comparison will need - is going to add overhead, both in terms of code mess and computation.

**Comment by Turmfalke2** [comment]
Just spend a hour trying to figure out why my ice was constantly melting. What is the reason for reservoirs to instantly melt nearby ice the moment you start to fill them? I am by no means an expert with vanilla mechanics, so connecting the proximity of a filled reservoir with the ice melting took a while.

**Comment by ReikaKalseki** [comment]
They apply their temperature, which is almost always above zero.

**Comment by Turmfalke2** [comment]
So I can use water to cool my crystallizer, but if I put it in a reservoir it is suddenly to hot?

**Comment by ReikaKalseki** [comment]
You can cool a reservoir a few degrees, but not below zero. You _do_ know that water freezes at zero, yes?

**Comment by Turmfalke2** [comment]
I am pretty sure in rotarycraft it takes -3 to freeze, but that is not the point. The temperature model is pretty simplistic, because with liquid water is always above zero, then why can I cool the lower the temperature of the freezer from -10 to like -25 by adding water?

**Comment by TomeWyrm** [comment]
You do know how a compressor works, right? The water is acting as a heat sink and radiator. The heat from the material getting chilled is transfered to the water, which then radiates it to the environment
